### PR TITLE
Clean rdmacm

### PR DESCRIPTION
**Pull request by rleon**

### CODE REVIEW

File: librdmacm/cma.c [hunk 1]
```diff
@@ -1526,6 +1526,8 @@ static void ucma_copy_conn_param_to_kern(struct cma_id_private *id_priv,
 
 int rdma_connect(struct rdma_cm_id *id, struct rdma_conn_param *conn_param)
 {
+	uint32_t qp_num = (conn_param) ? conn_param->qp_num : 0;
+	uint8_t srq = (conn_param) ? conn_param->srq : 0;
```
Comment (Hakon-Bugge) [inline librdmacm/cma.c]: Wy parentheses around conn_param?

File: librdmacm/cma.c [hunk 2]
```diff
@@ -1547,18 +1549,13 @@ int rdma_connect(struct rdma_cm_id *id, struct rdma_conn_param *conn_param)
...
-					     (id->qp->srq != NULL));
-	} else if (conn_param) {
-		ucma_copy_conn_param_to_kern(id_priv, &cmd.conn_param,
-					     conn_param, conn_param->qp_num,
-					     conn_param->srq);
-	} else {
-		ucma_copy_conn_param_to_kern(id_priv, &cmd.conn_param,
-					     conn_param, 0, 0);
+		qp_num = id->qp->qp_num;
+		srq = id->qp->srq != NULL;
```
Comment (Hakon-Bugge) [inline librdmacm/cma.c]: srq = !! id->qp->srq

File: librdmacm/cma.c (hunk 1 above)
Comment (rleon) [reply librdmacm/cma.c]: habbit, i can remove

File: librdmacm/cma.c (hunk 2 above)
Comment (rleon) [reply librdmacm/cma.c]: will change

**Comment by rleon** [comment]
Fixed, thanks

**Comment by rleon** [comment]
> Guess you had a small slip from your fingers.

Thanks, it is always the case when you need to do gazillion things in parallel.

**Comment by rleon** [comment]
> Very nitty, but if you look at the old code, you can drive the assertion that if(!id->qp) then conn_param != NULL (since it being de-referenced in the call to ucma_copy_conn_param_to_kern().
> 
> Hence I wonder, can we simply have:
> 
> uint32_t qp_num = id->qp ? id->qp->qp_num : conn_param->qp_num;
> (ditto for srq).
> and avoid the "if (id->qp)" below?

It is only partially true, the original code 

        if (id->qp) {
.....
       } else if (conn_param) {
...
       } else {
.... no QP and no conn_param
}

**Comment by Hakon-Bugge** [comment]
> > Very nitty, but if you look at the old code, you can drive the assertion that if(!id->qp) then conn_param != NULL (since it being de-referenced in the call to ucma_copy_conn_param_to_kern().
> > Hence I wonder, can we simply have:
> > uint32_t qp_num = id->qp ? id->qp->qp_num : conn_param->qp_num;
> > (ditto for srq).
> > and avoid the "if (id->qp)" below?
> 
> It is only partially true, the original code
> 
> ```
>     if (id->qp) {
> ```
> 
> .....
> } else if (conn_param) {
> ...
> } else {
> .... no QP and no conn_param
> }

In rdma_ connect(), yes. But in rdma_accept(), I only see if() .. else ..

**Comment by rleon** [comment]
I did it symmetrically, do you want me to change?

**Comment by Hakon-Bugge** [comment]
> I did it symmetrically, do you want me to change?

Yes. Keep it as simple as possible.

**Pull request merged**